### PR TITLE
Upgrade ESLint and associated dependency lib versions

### LINKED ISSUE #10105: Move from TSLint to ESLint
##### **Overview of the feature request**

TSLint will very soon move to maintenance mode and, starting from January 1st 2020 stop accepting anything except security fixes. This is a documented [issue](https://github.com/palantir/tslint/issues/4534), also explained in their [blog post](https://medium.com/palantir/tslint-in-2019-1a144c2317a9). I believe that JHipster should move from TSLint to ESLint in the similar timeframe.

##### **Motivation for or Use Case**

For sure TypeScript will continue to be developed, and hopefully JHipster and ESLint too. I think that staying stuck with the tool which will be abandoned soon is not an optimal way ahead.

Tasks:
- [x] Migrate angular code to use ESLint - #10187 
- [x] Migrate reactjs code to use ESLint - #10213 

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Closes #10105 

---
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
@jhipster/developers, This PR is ready for review.

File: generators/client/templates/angular/.eslintrc.json.ejs [hunk 1]
```diff
@@ -51,7 +60,7 @@
     "no-new": "error",
     "no-var": "error",
     "radix": "error",
-    "eqeqeq": ["error", "always", {"null": "ignore"}],
+    "eqeqeq": ["error", "always", { "null": "ignore"}],
```
Comment (MathieuAA) [inline generators/client/templates/angular/.eslintrc.json.ejs]: Unnecessary space

File: generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/auth.interceptor.ts.ejs [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class AuthInterceptor implements HttpInterceptor {
     }
 
     intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-        if (!request || !request.url || (/^http/.test(request.url) && !(SERVER_API_URL && request.url.startsWith(SERVER_API_URL)))) {
+        if (!request || !request.url || (request.url.startsWith('http') && !(SERVER_API_URL && request.url.startsWith(SERVER_API_URL)))) {
```
Comment (MathieuAA) [inline generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/auth.interceptor.ts.ejs]: why did you change it?

File: generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.ts.ejs [hunk 3]
```diff
@@ -71,8 +71,8 @@ export class <%=jhiPrefixCapitalized%>LoginModalComponent implements AfterViewIn
         }).then(() => {
             this.authenticationError = false;
             this.activeModal.dismiss('login success');
-            if (this.router.url === '/register' || (/^\/activate\//.test(this.router.url)) ||
-                (/^\/reset\//.test(this.router.url))) {
+            if (this.router.url === '/register' || this.router.url.startsWith('/activate/') ||
```
Comment (MathieuAA) [inline generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.ts.ejs]: why did you change it?

File: generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs [hunk 4]
```diff
@@ -37,7 +37,7 @@ export class Logout extends React.Component<ILogoutProps> {
     const logoutUrl = this.props.logoutUrl;
     if (logoutUrl) {
       // if Keycloak, logoutUrl has protocol/openid-connect in it
-      window.location.href = (logoutUrl.indexOf('/protocol') > -1) ?
+      window.location.href = logoutUrl.includes('/protocol') ?
```
Comment (MathieuAA) [inline generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs]: nice :)

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs [hunk 5]
```diff
@@ -216,8 +216,8 @@ _%>
     setFileData(event, field: string, isImage) {
         return new Promise((resolve, reject) => {
             if (event && event.target && event.target.files && event.target.files[0]) {
-                const file = event.target.files[0];
-                if (isImage && !/^image\//.test(file.type)) {
+                const file: File = event.target.files[0];
+                if (isImage && !file.type.startsWith('image/')) {
```
Comment (MathieuAA) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: why did you change it?

File: generators/client/templates/angular/src/main/webapp/app/layouts/profiles/profile.service.ts.ejs [hunk 6]
```diff
@@ -32,7 +32,7 @@ export class ProfileService {
     constructor(private http: HttpClient) { }
 
     getProfileInfo(): Promise<ProfileInfo> {
-        if (!this.profileInfo) {
+        if (this.profileInfo === undefined) {
```
Comment (MathieuAA) [inline generators/client/templates/angular/src/main/webapp/app/layouts/profiles/profile.service.ts.ejs]: if it's always undefined, okay. what about null? no chance of happening?

File: generators/client/templates/angular/tslint.json.ejs [hunk 7]
```diff
@@ -21,11 +21,6 @@
         "node_modules/codelyzer"
     ],
     "rules": {
-        "member-ordering": [
```
Comment (MathieuAA) [inline generators/client/templates/angular/tslint.json.ejs]: shouldn't the file be removed?

File: generators/client/templates/angular/.eslintrc.json.ejs (hunk 1 above)
Comment (vishal423) [reply generators/client/templates/angular/.eslintrc.json.ejs]: I believe `prettier` will fix that

File: generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/auth.interceptor.ts.ejs (hunk 2 above)
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/auth.interceptor.ts.ejs]: It reports an error due to typescript eslint [rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md) that's enabled by default as recommended configuration. I also prefer `startsWith` over regex in such scenarios as it's more readable.

File: generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.ts.ejs (hunk 3 above)
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.ts.ejs]: It reports an error due to typescript eslint [rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md) that's enabled by default as recommended configuration. I also prefer `startsWith` over regex in such scenarios as it's more readable.

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs (hunk 5 above)
Comment (vishal423) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs]: It reports an error due to typescript eslint [rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md) that's enabled by default as recommended configuration. I also prefer `startsWith` over regex in such scenarios as it's more readable.

File: generators/client/templates/angular/tslint.json.ejs (hunk 7 above)
Comment (vishal423) [reply generators/client/templates/angular/tslint.json.ejs]: no, we still need this to include `codelyzer` (angular specific) `tslint` rules.

File: generators/client/templates/angular/src/main/webapp/app/layouts/profiles/profile.service.ts.ejs (hunk 6 above)
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/layouts/profiles/profile.service.ts.ejs]: yes, I expect it to be always `undefined` as that's default Javascript variable value.

**Comment by vishal423** [comment]
@MathieuAA, do you have further feedback on this? Otherwise, I would like to merge this to proceed on angular dependency upgrade issue.

**Comment by vishal423** [comment]
I am merging this as I need this check-in before I upgrade other lib versions

**Comment by MathieuAA** [comment]
@vishal423 sorry got a lot on my plate. But yeah LGTM.
